src/frontend/src/instance/log_handler.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use api::v1::RowInsertRequests;
+use async_trait::async_trait;
+use auth::{PermissionChecker, PermissionCheckerRef, PermissionReq};
+use client::Output;
+use common_error::ext::BoxedError;
+use pipeline::{GreptimeTransformer, Pipeline};
+use servers::error::{
+    AuthSnafu, ExecuteGrpcRequestSnafu, PipelineSnafu, Result as ServerResult,
+    UnsupportedDeletePipelineSnafu,
+};
+use servers::query_handler::LogHandler;
+use session::context::QueryContextRef;
+use snafu::ResultExt;
+
+use super::Instance;
+
+#[async_trait]
+impl LogHandler for Instance {
+    async fn insert_logs(
+        &self,
+        log: RowInsertRequests,
+        ctx: QueryContextRef,
+    ) -> ServerResult<Output> {
+        self.plugins
+            .get::<PermissionCheckerRef>()
+            .as_ref()
+            .check_permission(ctx.current_user(), PermissionReq::LogWrite)
+            .context(AuthSnafu)?;
+
+        self.handle_log_inserts(log, ctx).await
+    }
```

Comment:
I think we only need `get_pipeline` as internal API here. Those CUD operations can be implemented via sql.

Reply:
We have a pipeline cache. If we use sql to manage those pipeline. we can not invalid the cache.

Reply:
We can enable TTL for the pipeline cache.

Reply:
replaced hashmap cache with moak with ttl.